pkg/artifact/downloader.go

```diff
@@ -65,13 +65,12 @@ func (l *downloader) Download(location *url.URL, verificationData auth.Verificat
 		return err
 	}
 
-	// rewind a second time to allow the archive to be unpacked
-	_, err = artifactFile.Seek(0, os.SEEK_SET)
+	err = artifactFile.Chmod(0644)
```

Comment:
one could also consider a chmod 0400 followed by a chown to the app user. or chmod 0640 and chgrp to the app user.